src/test/java/nl/tudelft/ewi/devhub/webtests/ProjectPullTest.java

```diff
@@ -0,0 +1,204 @@
+		commitApi = masterApi.getCommit();
+		commitModel = commitApi.get();
+	}
+
+	public void createBranch() throws URISyntaxException, GitAPIException, IOException {
+		Repository repository = repositoriesManager.getRepository(new URI(groupRepository.getRepositoryName() + ".git/"));
+		
+		Git git = Git.init().setBare(false).setDirectory(temporaryFolder.getRoot()).call();
+		
+		RemoteAddCommand addCommand = git.remoteAdd();
+		addCommand.setName("origin");
+		addCommand.setUri(new URIish(repository.getPath().toString()));
+		addCommand.call();
+		
+		git.pull().call();
+
+		git.checkout().setCreateBranch(true).setName(BRANCH_NAME).call();
+		Files.write(FILE_CONTENT.getBytes(), new File(temporaryFolder.getRoot(), FILE_NAME));
+		git.add().addFilepattern(FILE_NAME).call();
+		git.commit().setMessage(COMMIT_MESSAGE).setAuthor(user.getName(), user.getEmail()).setCommitter(user.getName(), user.getEmail()).call();
+		
+		git.push().call();
+	}
+
+	@Test
+	public void testCreatePullRequest() {
+		
+		// Assert branch is visible
+		Branch newBranch = openLoginScreen()
+			.login(NET_ID, PASSWORD)
+			.toCoursesView()
+			.listMyProjects()
+			.get(0).click()
+			.listBranches()
+			.get(1);
+
+
+		assertEquals(BRANCH_NAME, newBranch.getName());
+
+		// Navigate to pull request view
+		PullRequestOverViewView pullRequestOverViewView =
+			newBranch.click().openCreatePullRequestView();
+
+		// Wait for the view to load
+		try {
+			Thread.sleep(100);
+		} catch (InterruptedException e) {
+			log.warn("", e);
+		}
+
+		// Grab the pull request instance for BRANCH_NAME from the DB.
+		PullRequest pullRequest = getPullRequest(BRANCH_NAME);
+
+		// Assertions on pullRequestView against pullRequest
+		assertEquals(pullRequest.isOpen(), pullRequestOverViewView.isOpen());
+
+		String author = pullRequest.getDestination().getAuthor();
+		assertEquals(author, pullRequestOverViewView.getAuthorHeader());
+		
+		// Assert message header is latest commit message
+		assertEquals(COMMIT_MESSAGE, pullRequestOverViewView.getMessageHeader());
+
+	}
+
+	@Test
+	public void testClosePullRequest() {
+		// Assert branch is visible
+		Branch newBranch = openLoginScreen()
+				.login(NET_ID, PASSWORD)
+				.toCoursesView()
+				.listMyProjects()
+				.get(0).click()
+				.listBranches().get(1);
+
+		assertEquals(BRANCH_NAME, newBranch.getName());
+
+		// Navigate to pull request view
+		PullRequestOverViewView pullRequestOverViewView = newBranch.click().openCreatePullRequestView();
+		
+		assertTrue(pullRequestOverViewView.isOpen());
+		pullRequestOverViewView.close();
+		
+		// Wait for the view to load
+		try {
+			Thread.sleep(100);
+		} catch (InterruptedException e) {
+			log.warn("", e);
+		}
+		
+		assertFalse(pullRequestOverViewView.isOpen());
+		assertTrue(pullRequestOverViewView.isClosed());
+		assertFalse(pullRequestOverViewView.isMerged());
+		
+		PullRequest pullRequest = getPullRequest(BRANCH_NAME);
+		assertTrue(pullRequest.isClosed());
```

Comment:
While the assert statement is more readable, just a small reminder that `assertTrue` and `assertFalse` take an optional String parameter, which can be used for a more descriptive error message for the test runner. I do not ask you to change this, because for us it is useless, but you may see such a construct in the exam.